Add code

nchelluri opened the pull request.

Commit:
Add code

oalders reviewed (changes requested):
Great stuff. Just a couple of comments on the Docker side. I will now look at Vagrant.

Commit:
Fix grammar

Commit:
Fix sed commands

Commit:
Improvements per code review:

- if MM_ACCOUNT_ID or MM_LICENSE_KEY is not provided to Docker or to
  Vagrant, the process will abort
- use apt-get instead of apt to squelch warnings
- improve comment

Commit:
Security fixes

- Don't write sensitive vars to source code
  - Instead, pass them in in the environment
  - Also, get rid of sed code
- Escape input before using it

Commit:
Fix Vagrant trigger

Commit:
Tweak Dockerfile to make usage easier

Commit:
Do not escape IP addr passed to client library

oalders reviewed (approved):
💯
